frank/plot.py

```diff
@@ -31,308 +31,205 @@
-
-    ax.set_xlabel(r'Baseline [$\lambda$]')
-    ax.set_ylabel(ylabel)
-    ax.set_xscale(xscale)
-    ax.set_yscale(yscale)
-    ax.legend()
-
-    if zoom is not None:
-        ax.set_ylim(zoom)
-
-
-def plot_vis_resid(baselines, obs, fit, ax, c='k', marker='.', ls='None',
-                   binwidth='unspecified', xscale='log', yscale='linear',
-                   normalize_resid=False
-                   ):
+        ax.plot(baselines, vis, **kwargs)
+
+    ax.axhline(0, c='c', ls='--', zorder=10)
+
+    ax.legend(loc='best')
+
+
+def plot_vis_resid(baselines, resid, ax, **kwargs):
     r"""
     Plot residuals between visibility datapoints and a visibility domain fit
 
     Parameters
     ----------
     baselines : array
-        Baseline data coordinates. The assumed unit (for the x-label) is
-        :math:`\lambda`
-    obs : array
-        Observed visibility amplitude at baselines.
-        The assumed unit (for the y-label) is mJy
-    vis : array
-        Fitted visibility amplitude at baselines.
-        The assumed unit (for the y-label) is mJy
-    ax : Matplotlib axis
-        Axis on which to plot the residuals
-    c : Matplotlib color, default = 'k'
-        Color of residual points
-    marker : Matplotlib line style, default = '.'
-        Marker style of residual points
-    ls : Matplotlib line style, default = 'None'
-        Style of residual line
-    binwidth : int, float or string, default = 'unspecified'
-        Width of bins in which data is binned (for legend labels)
-    xscale : Matplotlib axis scale, default = 'log'
-        Scale for x-axis
-    yscale : Matplotlib axis scale, default = 'linear'
-        Scale for y-axis
-    normalize_resid : bool, default = False
-        Whether to plot the residuals normalized pointwise (i.e., locally)
-        by the data amplitude
+        Baseline data coordinates
+    resid : array
+        (Observed - fitted) visibility amplitudes at baselines
+    ax : Matplotlib `~.axes.Axes` class
+        Axis on which to plot
     """
 
-    resid = obs - fit
-    if normalize_resid:
-        resid /= obs
-    rmse = (np.mean(resid**2))**.5
-
-    ax.plot(baselines, resid, c=c, marker=marker, ls=ls,
-            label=r'{:.0f} k$\lambda$ bins, RMSE {:.3f}'.format(binwidth/1e3,
-                                                                rmse)
-            )
+    ax.plot(baselines, resid, **kwargs)
 
-    ax.set_xlabel(r'Baseline [$\lambda$]')
-    if normalize_resid:
-        ax.set_ylabel('Normalized\nresidual')
-    else:
-        ax.set_ylabel('Residual [mJy]')
-    ax.set_xscale(xscale)
-    ax.set_yscale(yscale)
-    ax.legend()
+    ax.legend(loc='best')
 
-    if yscale == 'linear':
-        ax.axhline(0, c='c', ls='--', zorder=10)
-        ax.set_ylim(-2 * rmse, 2 * rmse)
+    ax.axhline(0, c='c', ls='--', zorder=10)
 
 
-def plot_vis_hist(bin_edges, bin_counts, bin_width, ax, c): # TODO: just use hist
+def plot_vis_hist(bins, counts, ax, **kwargs):
```

Comment:
Why is this not using a UV Binner object?

Reply:
hm I think that's a bit overkill; this function is just a wrapper for `plt.hist`, simply taking in bins and counts. It's more generic than if it took in a `UVDataBinner` object. 

`plot_vis_hist` is also only called in `make_full_fig`, after `UVDataBinner` has already been called. `plot_vis_hist` uses the outputs of that call to `UVDataBinner`, but so do other functions in `make_full_fig`. I'd thus prefer not to have to call `UVDataBinner` again in `plot_vis_hist`.